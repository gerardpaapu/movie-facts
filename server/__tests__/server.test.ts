// @vitest-environment node
import { describe, it, expect } from 'vitest'
import { default as request } from 'supertest'

/**
 * This is not necessarily a test I'm usually keen to write, but
 * I'm just picking up easy coverage here and there
 */
describe('server wildcard rule', () => {
  it('serves index.html', async () => {
    process.env.NODE_ENV = 'production'
    const { default: server } = await import('../server')
    const res = await request(server).get('/movies')
    expect(res.status).toBe(200)
    expect(res.headers['content-type']).toMatch(/text\/html/)
    expect(res.text).toMatchSnapshot()
  })
})
