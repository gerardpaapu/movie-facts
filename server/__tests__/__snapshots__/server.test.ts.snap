// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`server wildcard rule > serves index.html 1`] = `
"<!DOCTYPE html>
<html lang=\\"en\\">
  <head>
    <meta charset=\\"UTF-8\\" />
    <title>React and web APIs!</title>
    <script type=\\"module\\" crossorigin src=\\"/assets/index-20551c5d.js\\"></script>
  </head>
  <body>
    <div id=\\"app\\"></div>
    
  </body>
</html>
"
`;
